src/utils/middleware.js

```diff
@@ -0,0 +1,24 @@
+const wrapTry = async (promise) => {
+  try {
+    return promise.then((res) => {
+      if (!res) {
+        return null;
+      }
+      if (!res.ok) throw new Error(`Request failed with ${res.status}`);
+      return res.json();
+    }).catch((error) => {
+      console.error(error);
+      return null;
+    });
+  } catch (err) {
+    return null;
+  }
+};
```

Comment:
This is also used in https://github.com/aeternity/superhero-ui/blob/1c36db32fd84292ccd7ee60b71e8a1bde799423a/src/utils/backend.js#L4-L22

Can it be extracted to a module/util.

Reply:
its not the same

Reply:
The only difference I see is the one in `backend.js` is commiting the backend status to store:
```javascript
store.commit('setBackendStatus', true); 
```

I still think this can be either generalized and extracted to module (possibly with conditional flag whether to store the backend status. Or if this side effect is negligible we can reuse it as is.

What do you suggest @thepiwo?

Reply:
6ba70e6be0b9fadaee765ca852b20a6280129a0b

Reply:
good solution, thanks